File: front/src/threejs/Scene/boundingBoxes.js
export class BallBoundingBox {
  #y_min;
  #y_max;
  #x_min;
  #x_max;

  constructor(player1Json, ballRadius) {
    const boardSize = player1Json['board']['size'];
    const halfRadius = ballRadius * 0.5;
    this.#y_max = boardSize['y'] * 0.5 - halfRadius;
    this.#y_min = -this.#y_max;
    this.#x_max = boardSize['x'] - halfRadius;
    this.#x_min = -this.#x_max;
  }

  get y_min() {
    return this.#y_min;
  }

  get y_max() {
    return this.#y_max;
  }

  get x_min() {
    return this.#x_min;
  }

  get x_max() {
    return this.#x_max;
  }
}

export class PaddleBoundingBox {
  #y_min;
  #y_max;

  constructor(player1Json) {
    const boardSize = player1Json['board']['size'];
    const paddleSize = player1Json['paddle']['size'];
    this.#y_max = boardSize['y'] * 0.5 - paddleSize['y'] * 0.5;
    this.#y_min = -this.#y_max;
  }

  get y_min() {
    return this.#y_min;
  }

  get y_max() {
    return this.#y_max;
  }
}

File: pong_server/src/game_server/Scene/bounding_boxes.py
import settings as settings


class BallBoundingBox(object):
    def __init__(self):
        half_radius = settings.BALL_RADIUS * 0.5
        self._y_max = settings.BOARD_SIZE[1] * 0.5 - half_radius
        self._y_min = -self._y_max
        self._x_max = settings.BOARD_SIZE[0] - half_radius
        self._x_min = -self._x_max

    def get_y_min(self):
        return self._y_min

    def get_y_max(self):
        return self._y_max

    def get_x_min(self):
        return self._x_min

    def get_x_max(self):
        return self._x_max


class PaddleBoundingBox(object):
    def __init__(self):
        self._y_max = settings.BOARD_SIZE[1] * 0.5 - settings.PADDLE_SIZE[1] * 0.5
        self._y_min = -self._y_max

    def get_y_min(self):
        return self._y_min

    def get_y_max(self):
        return self._y_max
